homeassistant/components/fyta/config_flow.py

```diff
@@ -0,0 +1,64 @@
+"""Config flow for FYTA integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from fyta_cli.fyta_connector import FytaConnector
+from fyta_cli.fyta_exceptions import (
+    FytaAuthentificationError,
+    FytaConnectionError,
+    FytaPasswordError,
+)
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+DATA_SCHEMA = vol.Schema(
+    {vol.Required(CONF_USERNAME): str, vol.Required(CONF_PASSWORD): str}
+)
+
+
+class FytaConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Fyta."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+
+        errors = {}
+        if user_input:
+            fyta = FytaConnector(user_input[CONF_USERNAME], user_input[CONF_PASSWORD])
+
+            try:
+                await fyta.login()
+                await fyta.client.close()
+            except FytaConnectionError:
+                errors["base"] = "cannot_connect"
+            except FytaAuthentificationError:
+                errors["base"] = "auth_error"
+            except FytaPasswordError:
+                errors["base"] = "auth_error"
+                errors[CONF_PASSWORD] = "password_error"
+            except Exception:  # pylint: disable=broad-except
+                errors["base"] = "unknown"
+            else:
+                self._async_abort_entries_match()
+
```

Comment:
We should place this after we got the user input. You can pass a dict for data to check. In that dict should be the username so we are at least doing our due dilligence on checking the account hasn't changed

Reply:
As in, we don't need to do a request to Fyta if we already know we have this entry, so we can move this to before that point